File: Web/RMA-Web-Shell/projects/shared-models-lib/src/lib/referrals/referral-search-request.ts
import { ModuleTypeEnum } from "../enums/module-type-enum";
import { PagedRequest } from "../pagination/PagedRequest";
import { ReferralItemTypeEnum } from "./referral-item-type-enum";
import { ReferralStatusEnum } from "./referral-status-enum";

export class ReferralSearchRequest {
    sourceModuleType: ModuleTypeEnum;
    targetModuleType: ModuleTypeEnum;
    assignedToRoleId: number;
    assignedByUserId: number;
    assignedToUserId: number;
    referralStatus: ReferralStatusEnum;
    referralItemType: ReferralItemTypeEnum;
    itemId: number;
    
    pagedRequest: PagedRequest;
}

File: Service/Admin/RMA.Service.Admin.MasterDataManager.Contracts/Entities/ReferralSearchRequest.cs

using RMA.Common.Entities.DatabaseQuery;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.Admin.MasterDataManager.Contracts.Entities
{
    public class ReferralSearchRequest
    {
        public ModuleTypeEnum? SourceModuleType { get; set; }
        public ModuleTypeEnum? TargetModuleType { get; set; }
        public int? AssignedToRoleId { get; set; }
        public int? AssignedByUserId { get; set; }
        public int? AssignedToUserId { get; set; }
        public ReferralStatusEnum? ReferralStatus { get; set; }
        public ReferralItemTypeEnum? ReferralItemType { get; set; }
        public int? ItemId { get; set; }
        public PagedRequest PagedRequest { get; set; }
    }
}